tests/queries/0_stateless/02943_variant_read_subcolumns.sh

```diff
@@ -7,8 +7,7 @@ CLICKHOUSE_LOG_COMMENT=
 # shellcheck source=../shell_config.sh
 . "$CUR_DIR"/../shell_config.sh
 
-CH_CLIENT="$CLICKHOUSE_CLIENT --allow_experimental_variant_type=1 --use_variant_as_common_type=1 --allow_suspicious_variant_types=1 --max_insert_threads 4 --group_by_two_level_threshold 752249 --group_by_two_level_threshold_bytes 15083870 --distributed_aggregation_memory_efficient 1 --fsync_metadata 1 --output_format_parallel_formatting 0 --input_format_parallel_parsing 0 --min_chunk_bytes_for_parallel_parsing 6583861 --max_read_buffer_size 640584 --prefer_localhost_replica 1 --max_block_size 38844 --max_threads 48 --optimize_append_index 0 --optimize_if_chain_to_multiif 1 --optimize_if_transform_strings_to_enum 0 --optimize_read_in_order 1 --optimize_or_like_chain 0 --optimize_substitute_columns 1 --enable_multiple_prewhere_read_steps 1 --read_in_order_two_level_merge_threshold 4 --optimize_aggregation_in_order 0 --aggregation_in_order_max_block_bytes 18284646 --use_uncompressed_cache 1 --min_bytes_to_use_direct_io 10737418240 --min_bytes_to_use_mmap_io 10737418240 --local_filesystem_read_method pread --remote_filesystem_read_method read --local_filesystem_read_prefetch 1 --filesystem_cache_segments_batch_size 0 --read_from_filesystem_cache_if_exists_otherwise_bypass_cache 0 --throw_on_error_from_cache_on_write_operations 1 --remote_filesystem_read_prefetch 0 --allow_prefetched_read_pool_for_remote_filesystem 0 --filesystem_prefetch_max_memory_usage 128Mi --filesystem_prefetches_limit 0 --filesystem_prefetch_min_bytes_for_single_read_task 16Mi --filesystem_prefetch_step_marks 50 --filesystem_prefetch_step_bytes 0 --compile_aggregate_expressions 1 --compile_sort_description 0 --merge_tree_coarse_index_granularity 31 --optimize_distinct_in_order 1 --max_bytes_before_external_sort 1 --max_bytes_before_external_group_by 1 --max_bytes_before_remerge_sort 2640239625 --min_compress_block_size 3114155 --max_compress_block_size 226550 --merge_tree_compact_parts_min_granules_to_multibuffer_read 118 --optimize_sorting_by_input_stream_properties 0 --http_response_buffer_size 543038 --http_wait_end_of_query False --enable_memory_bound_merging_of_aggregation_results 1 --min_count_to_compile_expression 3 --min_count_to_compile_aggregate_expression 3 --min_count_to_compile_sort_description 0 --session_timezone America/Mazatlan --prefer_warmed_unmerged_parts_seconds 8 --use_page_cache_for_disks_without_file_cache False --page_cache_inject_eviction True --merge_tree_read_split_ranges_into_intersecting_and_non_intersecting_injection_probability 0.82 "
-
+CH_CLIENT="$CLICKHOUSE_CLIENT --allow_experimental_variant_type=1 --use_variant_as_common_type=1 --allow_suspicious_variant_types=1"
```

Comment:
Is it intended to fix `Checksum doesn't match` with removing the settings?
It is because `remote_filesystem_read_method=read` which is broken?



<details>
<summary>Example</summary>

```
2024-05-25 09:43:20 [ecf9170fca01] 2024.05.25 05:42:56.459353 [ 4580 ] {d156a51d-50c5-427a-bb98-9100706b7877} <Error> executeQuery: Code: 40. DB::Exception: Checksum doesn't match: corrupted data. Reference: e7617c3cc3d3632496ea14f633a22c91. Actual: 08931924daca90b17d95d704f29dfef0. Size of compressed block: 100141: (while reading column id): (while reading from part store/4de/4debd4d4-4d47-4455-b9a1-6e074cad398d/all_10_10_0/ in table test_17icn9m2.test (4debd4d4-4d47-4455-b9a1-6e074cad398d) located on disk s3_with_keeper of type s3, from mark 0 with max_rows_to_read = 63603): While executing MergeTreeSelect(pool: ReadPoolInOrder, algorithm: InOrder). (CHECKSUM_DOESNT_MATCH) (version 24.6.1.1 (official build)) (from [::1]:57904) (in query: select v, v.UInt64, v.`Array(Variant(String, UInt64))`, v.`Array(Variant(String, UInt64))`.size0, v.`Array(Variant(String, UInt64))`.UInt64 from test order by id format Null), Stack trace (when copying this message, always include the lines below):
2024-05-25 09:43:20 
2024-05-25 09:43:20 0. ./contrib/llvm-project/libcxx/include/exception:141: Poco::Exception::Exception(String const&, int) @ 0x0000000032aa7e74
2024-05-25 09:43:20 1. ./build_docker/./src/Common/Exception.cpp:101: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000001954ba4b
2024-05-25 09:43:20 2. ./contrib/llvm-project/libcxx/include/string:1499: DB::Exception::createDeprecated(String const&, int, bool) @ 0x000000001963bf70
2024-05-25 09:43:20 3. ./build_docker/./src/Compression/CompressedReadBufferBase.cpp:0: DB::validateChecksum(char*, unsigned long, CityHash_v1_0_2::uint128) @ 0x000000002656825c
2024-05-25 09:43:20 4. ./build_docker/./src/Compression/CompressedReadBufferBase.cpp:206: DB::CompressedReadBufferBase::readCompressedData(unsigned long&, unsigned long&, bool) @ 0x0000000026566f6f
2024-05-25 09:43:20 5. ./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:815: DB::CachedCompressedReadBuffer::nextImpl() @ 0x000000002ba0239d
2024-05-25 09:43:20 6. DB::ReadBuffer::next() @ 0x000000000b58e4eb
2024-05-25 09:43:20 7. DB::ReadBuffer::readBig(char*, unsigned long) @ 0x000000000bb0a6a1
2024-05-25 09:43:20 8. ./build_docker/./src/DataTypes/Serializations/SerializationNumber.cpp:222: DB::SerializationNumber<unsigned long>::deserializeBinaryBulk(DB::IColumn&, DB::ReadBuffer&, unsigned long, double) const @ 0x00000000269197d0
2024-05-25 09:43:20 9. ./build_docker/./src/DataTypes/Serializations/ISerialization.cpp:0: DB::ISerialization::deserializeBinaryBulkWithMultipleStreams(COW<DB::IColumn>::immutable_ptr<DB::IColumn>&, unsigned long, DB::ISerialization::DeserializeBinaryBulkSettings&, std::shared_ptr<DB::ISerialization::DeserializeBinaryBulkState>&, std::unordered_map<String, COW<DB::IColumn>::immutable_ptr<DB::IColumn>, std::hash<String>, std::equal_to<String>, std::allocator<std::pair<String const, COW<DB::IColumn>::immutable_ptr<DB::IColumn>>>>*) const @ 0x00000000268461d9
2024-05-25 09:43:20 10. ./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:701: DB::MergeTreeReaderCompact::readData(DB::NameAndTypePair const&, COW<DB::IColumn>::immutable_ptr<DB::IColumn>&, unsigned long, std::function<DB::ReadBuffer* (DB::ISerialization::SubstreamPath const&)> const&) @ 0x000000002b9ebb68
2024-05-25 09:43:20 11. ./contrib/llvm-project/libcxx/include/__functional/function.h:818: ? @ 0x000000002b9f763e
2024-05-25 09:43:20 12. ./build_docker/./src/Storages/MergeTree/MergeTreeRangeReader.cpp:112: DB::MergeTreeRangeReader::DelayedStream::finalize(std::vector<COW<DB::IColumn>::immutable_ptr<DB::IColumn>, std::allocator<COW<DB::IColumn>::immutable_ptr<DB::IColumn>>>&) @ 0x000000002b9b3eca
2024-05-25 09:43:20 13. ./build_docker/./src/Storages/MergeTree/MergeTreeRangeReader.cpp:0: DB::MergeTreeRangeReader::startReadingChain(unsigned long, DB::MarkRanges&) @ 0x000000002b9ca9ee
2024-05-25 09:43:20 14. ./build_docker/./src/Storages/MergeTree/MergeTreeRangeReader.cpp:0: DB::MergeTreeRangeReader::read(unsigned long, DB::MarkRanges&) @ 0x000000002b9c625b
2024-05-25 09:43:20 15. ./build_docker/./src/S
```

</details>

Reply:
> Is it intended to fix Checksum doesn't match with removing the settings?

Mostly because I just forgot to remove them after testing, but it also fixes `Checksum doesn't match`, yes.

> It is because remote_filesystem_read_method=read which is broken? Shouldn't we at least set remote_filesystem_read_method to have it non randomized?

Yes, because of broken `remote_filesystem_read_method=read`. It's broken only in private and we disabled randomization of this setting in private, so no need to set it.